File: apps/agents/src/astra/deps.py
from dataclasses import dataclass, field
from pathlib import Path


@dataclass(frozen=True)
class AstraDeps:
    """Shared dependency context passed to Astra agents and tools."""

    user_id: str
    workspace: str
    conversation_id: str | None = None
    research_artifact_dir: Path | None = None
    constraints: tuple[str, ...] = field(default_factory=tuple)

    def describe(self) -> str:
        constraints = "; ".join(self.constraints) if self.constraints else "none"
        conversation_id = self.conversation_id or "none"
        artifact_dir = (
            self.research_artifact_dir.as_posix() if self.research_artifact_dir else "none"
        )
        return (
            f"user_id={self.user_id}; workspace={self.workspace}; "
            f"conversation_id={conversation_id}; research_artifact_dir={artifact_dir}; "
            f"constraints={constraints}"
        )

File: apps/api/src/agents/deps.ts
export type AstraDeps = {
  userId: string;
  workspace: string;
  conversationId?: string;
  runId?: string;
  constraints: string[];
};

export function describeAstraDeps(deps: AstraDeps): string {
  const constraints = deps.constraints.length > 0 ? deps.constraints.join("; ") : "none";
  const conversationId = deps.conversationId?.trim() || "none";
  const runId = deps.runId?.trim() || "none";

  return `user_id=${deps.userId}; workspace=${deps.workspace}; conversation_id=${conversationId}; run_id=${runId}; constraints=${constraints}`;
}
